src/integrations/kafka/consumer.cpp

```diff
@@ -339,7 +345,22 @@ void Consumer::StartConsuming() {
 
       try {
         consumer_function_(batch);
-        if (const auto err = consumer_->commitSync(); err != RdKafka::ERR_NO_ERROR) {
+        std::vector<RdKafka::TopicPartition *> partitions;
+        utils::OnScopeExit clear_partitions([&p = partitions]() { RdKafka::TopicPartition::destroy(p); });
+
+        if (const auto err = consumer_->assignment(partitions); err != RdKafka::ERR_NO_ERROR) {
+          spdlog::warn("Saving the commited offset of consumer {} failed: {}", info_.consumer_name,
+                       RdKafka::err2str(err));
+          throw ConsumerCheckFailedException(
+              info_.consumer_name,
+              fmt::format("Couldn't get assignment to commit offsets: '{}'", RdKafka::err2str(err)));
+        }
+
+        for (auto *partition : partitions) {
+          partition->set_offset(batch.back().Offset() + 1);
```

Comment:
Why don't we store the `batch.back().Offset()` into a variable?

Reply:
Because it's not used anywhere else?

Reply:
I was going more for getting the same value from the back of the vector for each partition instead of getting it once and storing it into a variable.